### PR TITLE
Fix grad accumulation

### PR DESCRIPTION
**Pull request by younesbelkada**
Fixes https://github.com/lvwerra/trl/issues/583

As reported by @Uncomfy, the tests used to pass within a single training step but fails if one uses larger dataset - leading to the gradient accumulation not working properly. 

Indeed one needs to perform gradient accumulation as showed in the accelerate tutorial

I propose to merge this PR as a hotfix before waiting for https://github.com/lvwerra/trl/pull/546 to get merged.

cc @lvwerra @edbeeching

### CODE REVIEW

**Review by edbeeching — APPROVED** [review]
Thanks for fixing this, I mentioned it the other day but it makes sense to fix asap.
LGTM

File: tests/test_ppo_trainer.py [hunk 1]
```diff
@@ -1151,7 +1150,6 @@ def test_grad_accumulation(self):
             reward = [torch.tensor(1.0), torch.tensor(1.0)]
             # train model by running a step twice
             _ = ppo_trainer.step([q for q in query_tensor], [r for r in response_tensor], reward)
-            break
 
         model_grad_acc = gpt2_model_clone.v_head.summary.weight.grad.clone()
         self.assertTrue(torch.allclose(model_grad_acc, model_grad, rtol=1e-3, atol=1e-3))
```
Comment (vwxyzjn) [inline tests/test_ppo_trainer.py:1155]: A quick comment here: at some point I tried printing out the grads and they are all zeros here — I think it’s because we already did `optimizer.step`. We should be comparing the weights instead: 

```
self.assertTrue(torch.allclose(gpt2_model_clone.v_head.summary.weight, gpt2_model.v_head.summary.weight, rtol=1e-3, atol=1e-3))
```

**Review by lvwerra — APPROVED** [review]

File: tests/test_ppo_trainer.py (hunk 1 above)
Comment (younesbelkada) [reply tests/test_ppo_trainer.py:1155]: Thanks, added that suggestion !
